server/src/main/java/org/opensearch/rest/action/admin/cluster/RestClusterDeleteWeightedRoutingAction.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rest.action.admin.cluster;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.admin.cluster.shards.routing.weighted.delete.ClusterDeleteWeightedRoutingRequest;
+import org.opensearch.client.Requests;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.action.RestToXContentListener;
+
+import java.io.IOException;
+import java.util.List;
+
+import static java.util.Collections.singletonList;
+import static org.opensearch.rest.RestRequest.Method.DELETE;
+
+/**
+ * Delete Weighted Round Robin based shard routing weights
+ *
+ * @opensearch.api
+ *
+ */
+public class RestClusterDeleteWeightedRoutingAction extends BaseRestHandler {
+
+    private static final Logger logger = LogManager.getLogger(RestClusterDeleteWeightedRoutingAction.class);
+
+    @Override
+    public List<Route> routes() {
+        return singletonList(new Route(DELETE, "/_cluster/routing/awareness/weights"));
+    }
```

Comment:
do we want to support explicit attribute deletion?

Reply:
yeah, but this will require few changes in the way data is persisted in cluster state as well. Create an issue to work on this https://github.com/opensearch-project/OpenSearch/issues/4747